app/src/main/java/org/fossasia/pslab/adapters/ControlMainAdapter.java

```diff
@@ -13,17 +17,37 @@
  * Created by asitava on 6/6/17.
  */
 
-public class ControlMainAdapter extends RecyclerView.Adapter<ControlMainAdapter.ViewHolder> {
+public class ControlMainAdapter extends RecyclerView.Adapter<ControlMainAdapter.ViewHolder> implements View.OnClickListener{
```

Comment:
If you are setting ```OnClickListener``` individually to all view elements in ```onBindViewHolder```, then why implement ```OnClickListener``` to complete adapter.